packages/conform-react/helpers.ts

```diff
@@ -60,24 +70,37 @@ type InputOptions = BaseOptions &
 		| {
 				type: 'checkbox' | 'radio';
 				value?: string;
+				id?: string;
 		  }
 		| {
-				type?: Exclude<HTMLInputTypeAttribute, 'button' | 'submit' | 'hidden'>;
+				type?: Exclude<InputProps['type'], 'button' | 'submit' | 'hidden'>;
 				value?: never;
+				id?: never;
 		  }
 	);
 
-function getFormControlProps(
-	config: FieldConfig<any>,
+function getFormControlProps<
+	Schema,
+	Tag extends 'input' | 'select' | 'textarea',
+>(
+	config: FieldConfig<Schema>,
+	_tag: Tag,
```

Comment:
I do not follow why we need to change the result type of this function. It looks strange to pass an argument only as a type hint.